tests/convergence-test.js

```diff
@@ -532,7 +532,24 @@ describe('BigTest Convergence', () => {
         expect(stats.runs).to.be.within(8, 12);
         expect(stats.timeout).to.equal(100);
         expect(stats.value).to.equal(50);
-        expect(stats.stack).to.have.lengthOf(4);
+        expect(stats.queue).to.have.lengthOf(4);
+      });
+
+      it('stores and calls the callbacks first-in-first-out', () => {
+        let arr = [];
+
+        return converge
+          .when(() => true)
+          .do(() => {
+            arr.push('first');
+          })
+          .do(() => {
+            arr.push('second');
+          })
+          .do(() => {
+            expect(arr.join(', ')).to.equal('first, second');
+          })
+          .run();
```

Comment:
The other tests here are written in a specific way, so we should try to be consistent when adding new ones. The main thing is: we don't want to put the real assertions in the convergence to ensure that we don't get a false positive in the off chance that assertion doesn't get called.

We also don't need `.when(() => true)` here, and can use `.to.deep.equal` instead of joining the array.

```js
it('stores and calls the callbacks first-in-first-out', async () => {
  let arr = [];
  
  let assertion = converge
    .do(() => arr.push('first'))
    .do(() => arr.push('second'));

  await expect(assertion.run()).to.be.fulfilled;
  expect(arr).to.deep.equal(['first', 'second']);
});
```

Reply:
K, updated. 😃